File: src/main/java/com/RestauCentral/RestauCentral/Service/OffreService.java
package com.RestauCentral.RestauCentral.Service;


import com.RestauCentral.RestauCentral.Model.Offre;
import java.util.List;

public interface OffreService {
    List<Offre> getAllOffres();
    Offre getOffreById(Long id);
    Offre addOffre(Offre offre);
    void deleteOffre(Long id);
    Offre updateOffre(Long id, Offre offre);
    List<Offre> getOffresByRestaurantId(Long restaurantId);
}

File: restau-central-frontend/src/app/services/OffreService.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Offre } from './offre.model';

@Injectable({
  providedIn: 'root'
})
export class OffreService {
  private baseUrl = 'http://localhost:8080/offres';

  constructor(private http: HttpClient) {}

  getOffresByRestaurantId(restaurantId: number): Observable<Offre[]> {
    return this.http.get<Offre[]>(`${this.baseUrl}/restaurant/${restaurantId}`);
  }

  getOffreById(id: number): Observable<Offre> {
    return this.http.get<Offre>(`${this.baseUrl}/${id}`);
  }
}
